packages/firestore/test/unit/remote/node/serializer.test.ts

```diff
@@ -693,7 +696,9 @@ describe('Serializer', () => {
           value: { stringValue: 'food' }
         }
       });
-      expect(s.fromFieldFilter(actual)).to.deep.equal(input);
+      const roundtripped = s.fromFieldFilter(actual);
+      expect(roundtripped).to.deep.equal(input);
+      expect(roundtripped).to.be.instanceof(FieldFilter);
```

Comment:
FWIW we could instead do:
```
expect(Object.getPrototypeOf(roundtripped)).to.equal(Object.getPrototypeOf(input));
```
to avoid the redundancy of specifying the type of `input`...  but in practice it doesn't matter and it's easier to read as-is... so probably not a good improvement (unless we wanted to refactor these tests to share code or something).

Reply:
I don't know how `getPrototypeOf` works -- if it checks for the _exact_ class, then it could be an improvement, because given, say, `ArrayContainsFilter`, it will nevertheless satisfy a check for `instanceof FieldFilter` (due to inheritance).

Reply:
Ah, yes.  I believe the prototype check would be more strict than that...  that said it's not super normal to deal with an object's prototype directly or do equal checks on it.  I am not 100% confident it wouldn't bite us in the future some day (due to browser differences or something).  So I don't feel strongly that it's a better approach or anything (i.e. feel free to leave as-is).